service/src/main/java/com/commercetools/util/SphereClientConfigurationUtil.java

```diff
@@ -57,11 +59,17 @@ protected static BlockingSphereClient decorateSphereClient(@Nonnull final Sphere
     private static SphereClient withRetry(@Nonnull final SphereClient sphereClient) {
         final RetryAction scheduledRetry =
                 RetryAction.ofScheduledRetry(RETRIES_LIMIT, context -> calculateVariableDelay());
+
+        final RetryPredicate httpErrorMatcher =
+                RetryPredicate.ofMatchingErrors(io.sphere.sdk.http.HttpException.class);
+
         final RetryPredicate http5xxMatcher =
-                RetryPredicate.ofMatchingStatusCodes(
-                        BAD_GATEWAY_502, SERVICE_UNAVAILABLE_503, GATEWAY_TIMEOUT_504);
+                RetryPredicate.ofMatchingStatusCodes(BAD_GATEWAY_502, SERVICE_UNAVAILABLE_503, GATEWAY_TIMEOUT_504);
+
+        final Predicate<RetryContext> retryPredicate = http5xxMatcher.or(httpErrorMatcher);
```

Comment:
chaining the exceptionMatcher with our normal HTTP retry will cause to add variable delay for both matches. If we have this random Exception happening, the customer will wait 10 seconds to complete the payment. I would not add any delay if it is a HttpException. All we want to make sure is that, if there is a HttpException, we should retry our request in a different connection from the HTTP connection pool.

Reply:
https://github.com/commercetools/commercetools-payone-integration/pull/329/commits/d74bb7c925f1a3371f4078f61fbe4847c9dc3e3a